src/Microsoft.ML.Ensemble/Trainer/EnsembleTrainerBase.cs

```diff
@@ -190,15 +179,17 @@ private void TrainCore(IChannel ch, RoleMappedData data)
                 if (stackingTrainer != null)
                     stackingTrainer.Train(modelsList, _subsetSelector.GetTestData(null, batch), Host);
 
-                foreach (var model in modelsList)
-                    Utils.Add(ref Models, model);
+                Models.AddRange(modelsList);
                 int modelSize = Utils.Size(Models);
                 if (modelSize < Utils.Size(Trainers))
                     ch.Warning("{0} of {1} trainings failed.", Utils.Size(Trainers) - modelSize, Utils.Size(Trainers));
                 ch.Check(modelSize > 0, "Ensemble training resulted in no valid models.");
             }
+            return CreatePredictor(Models);
         }
 
+        protected internal abstract TPredictor CreatePredictor(List<FeatureSubsetModel<IPredictorProducing<TOutput>>> models);
```

Comment:
Similar thoughts here - are we expecting other callers from within this assembly to call this method?  I think it should just be marked `protected` and not `protected internal`.

NOTE: If you want the intersection between `protected` and `internal` (meaning only sub-classes that are in this assembly can see this method), then you are looking for a new C# 7.2 feature: `private protected`.  See [https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/private-protected](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/private-protected)

Reply:
Yeah, that's what I wanted, thanks @eerhardt !